tests/test_tools.py

```diff
@@ -209,3 +211,156 @@ def __init__(self, value: int):
         # Cache should be cleared after Operator construction
         cache_size = Object._instance_cache.cache_info()[-1]
         assert cache_size == 0
+
+
+class TestMemoizedMethods:
```

Comment:
Do these tests get specifically run in the GIL-less CI?

Reply:
You should also probably have some tests to make sure this doesn't do anything weird when using `_rebuild`